diff --git a/util.c b/util.c
index 85366bf..bbc57f3 100644
--- a/util.c
+++ b/util.c
@@ -99,10 +99,10 @@ fmt_human(uintmax_t num, int base)
 	double scaled;
 	size_t i, prefixlen;
 	const char **prefix;
-	const char *prefix_1000[] = { "", "k", "M", "G", "T", "P", "E", "Z",
-	                              "Y" };
-	const char *prefix_1024[] = { "", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei",
-	                              "Zi", "Yi" };
+	const char *prefix_1000[] = { "", " k", " M", " G", " T", " P", " E", " Z",
+	                              " Y" };
+	const char *prefix_1024[] = { "", " Ki", " Mi", " Gi", " Ti", " Pi", " Ei",
+	                              " Zi", " Yi" };

 	switch (base) {
 	case 1000:
@@ -123,7 +123,7 @@ fmt_human(uintmax_t num, int base)
 		scaled /= base;
 	}

-	return bprintf("%.1f %s", scaled, prefix[i]);
+	return bprintf("%.1f%s", scaled, prefix[i]);
 }

 int
